src/handlers/handleCurrentPage/handleCurrentPage.tsx

```diff
@@ -0,0 +1,37 @@
+interface CurrentPageHandler {
+  event: React.MouseEvent<HTMLAnchorElement>;
+  currentPage: number;
+  pageItemsCount: number;
+  setCurrentPage: (pageNumber: number) => void;
+  newParams: URLSearchParams;
+  setSearchParams: (currentOptions: URLSearchParams) => void;
+}
+
+export default function handleCurrentPage(params: CurrentPageHandler): void {
+  const {
+    event,
+    currentPage,
+    pageItemsCount,
+    setCurrentPage,
+    newParams,
+    setSearchParams,
+  } = params;
+  const value = (event.target as HTMLAnchorElement).text;
```

Comment:
The `value` is extracted from the `event.target` assuming it is an `HTMLAnchorElement`. Ensure that this assumption holds true in all cases where this handler is used, as it might lead to runtime errors if `event.target` is not an anchor element.